From 7e8b6c3551ea6a2b4e9201188a5fb79c5970f114 Mon Sep 17 00:00:00 2001
From: dobatog <dobatog@gmail.com>
Date: Thu, 26 Feb 2015 12:32:27 +0100
Subject: [PATCH 6/9] ARM: TLV320AIC23 SoC Audio Codec: Fix errors reported
 related to input routing signals.

The following patch, based on stable v3.19,  corrects the errors that are reported in the boot trace in reference to the Input Signals of the audio codec TLV320AIC23:

tlv320aic23-codec 1-001a: Control not supported for path LLINEIN -> [NULL] -> Line Input
tlv320aic23-codec 1-001a: ASoC: no dapm match for LLINEIN --> NULL --> Line Input
tlv320aic23-codec 1-001a: ASoC: Failed to add route LLINEIN -> NULL -> Line Input
tlv320aic23-codec 1-001a: Control not supported for path RLINEIN -> [NULL] -> Line Input
tlv320aic23-codec 1-001a: ASoC: no dapm match for RLINEIN --> NULL --> Line Input
tlv320aic23-codec 1-001a: ASoC: Failed to add route RLINEIN -> NULL -> Line Input
tlv320aic23-codec 1-001a: Control not supported for path MICIN -> [NULL] -> Mic Input
tlv320aic23-codec 1-001a: ASoC: no dapm match for MICIN --> NULL --> Mic Input
tlv320aic23-codec 1-001a: ASoC: Failed to add route MICIN -> NULL -> Mic Input

I am trying to set the sound system in a CM-510 (Compulab - SoM) based board with DT, using Simple-Card-Audio and  the TLV320AIC23 audio codec included in the SoM . I faced this problem and thanks to the help of Sebastian Hesselbarth who noticed the possible error located in the structure snd_soc_dapm_route tlv320aic23_intercon[] , now errors are not reported.

Tested on CM-510 (Compulab SoM Board).

Signed-off-by: dobatog <dobatog@gmail.com>
Tested-by: Max Filippov <jcmvbkbc@gmail.com>
---
 sound/soc/codecs/tlv320aic23.c | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/sound/soc/codecs/tlv320aic23.c b/sound/soc/codecs/tlv320aic23.c
index cd8c02b..95e6b11 100644
--- a/sound/soc/codecs/tlv320aic23.c
+++ b/sound/soc/codecs/tlv320aic23.c
@@ -174,10 +174,10 @@ static const struct snd_soc_dapm_route tlv320aic23_intercon[] = {
 	{"ROUT", NULL, "Output Mixer"},
 
 	/* Inputs */
-	{"Line Input", "NULL", "LLINEIN"},
-	{"Line Input", "NULL", "RLINEIN"},
+	{"Line Input", NULL, "LLINEIN"},
+	{"Line Input", NULL, "RLINEIN"},
 
-	{"Mic Input", "NULL", "MICIN"},
+	{"Mic Input", NULL, "MICIN"},
 
 	/* input mux */
 	{"Capture Source", "Line", "Line Input"},
-- 
2.6.4

